tests/phpunit/tests/option/transient.php

```diff
@@ -184,4 +184,77 @@ public function test_nonexistent_key_old_timeout() {
 		);
 		$this->assertSame( $expected, $a->get_events() );
 	}
+
+	/**
+	 * @ticket 58903
+	 *
+	 * @dataProvider data_transient_keys
+	 *
+	 * @covers ::set_transient()
+	 */
+	public function test_transient_key_valid( $key, $expected, $message ) {
+		$value = rand_str();
+
+		$this->assertSame( $expected, set_transient( $key, $value ), $message );
+	}
+
+	/**
+	 * @ticket 58903
+	 *
+	 * @covers ::set_transient()
+	 *
+	 * @return array[] {
+	 *    mixed  Arguments used as transient name/key
+	 *    bool   Whether the transient key is expected to redirect
+	 *    string Error value when condition test fails
+	 * }
+	 */
+	public function data_transient_keys() {
+		return array(
+			array(
+				'',
+				false,
+				'set_transient() should not accept empty string values for the transient name.',
```

Comment:
Instead of (or in addition to) providing the description like this, consider using array keys to make the cases more descriptive.

See https://docs.phpunit.de/en/10.3/writing-tests-for-phpunit.html#writing-tests-for-phpunit-data-providers-examples-nameddatasetstest-php